### PR TITLE
fix: antd ConfigProvider not found

### LINKED ISSUE #2990: umi-plugin-locale not support antd 3.9.x
image

There's no `ConfigProvider` in antd 3.9.x.

### PR DESCRIPTION
**Pull request by ycjcl868**
Close #2990.

##### Checklist

- [ ] `npm test` passes
- [ ] tests are included
- [ ] documentation is changed or added
- [ ] commit message follows commit guidelines

##### Description of change

- any feature?
- close https://github.com/umijs/umi/ISSUE_URL

### CODE REVIEW

File: packages/umi-plugin-locale/template/wrapper.jsx.tpl [hunk 1]
```diff
@@ -122,9 +122,20 @@ class LocaleWrapper extends React.Component{
...
     {{#antd}}
-     return (<LocaleProvider locale={appLocale.antd ? (appLocale.antd.default || appLocale.antd) : defaultAntd}>
+     // avoid antd ConfigProvider not found
+     let AntdProvider = LocaleProvider;
+     const [major, minor] = `${version || ''}`.split('.');
+     // antd 3.21.0 use ConfigProvider not LocaleProvider
+     const isConfigProvider = Number(major) > 3 || (Number(major) >= 3 && Number(minor) >= 21);
+     if (isConfigProvider) {
+       try {
+         AntdProvider = require('antd/lib/config-provider').default;
```
Comment (sorrycc) [inline packages/umi-plugin-locale/template/wrapper.jsx.tpl]: 用 `antd/es/config-provide` ?

Comment (ycjcl868) [reply packages/umi-plugin-locale/template/wrapper.jsx.tpl]: ssr 会报错

File: packages/umi-plugin-locale/template/wrapper.jsx.tpl [hunk 2]
```diff
@@ -122,9 +122,20 @@ class LocaleWrapper extends React.Component{
     </IntlProvider>)
     {{/localeList.length}}
     {{#antd}}
-     return (<LocaleProvider locale={appLocale.antd ? (appLocale.antd.default || appLocale.antd) : defaultAntd}>
+     // avoid antd ConfigProvider not found
+     let AntdProvider = LocaleProvider;
+     const [major, minor] = `${version || ''}`.split('.');
+     // antd 3.21.0 use ConfigProvider not LocaleProvider
+     const isConfigProvider = Number(major) > 3 || (Number(major) >= 3 && Number(minor) >= 21);
```
Comment (afc163) [inline packages/umi-plugin-locale/template/wrapper.jsx.tpl]: 可以用 https://www.npmjs.com/package/semver ？

Comment (ycjcl868) [reply packages/umi-plugin-locale/template/wrapper.jsx.tpl]: 主要是不想再引个包

File: packages/umi-plugin-locale/template/wrapper.jsx.tpl (hunk 1 above)
Comment (afc163) [reply packages/umi-plugin-locale/template/wrapper.jsx.tpl]: same problem: https://github.com/ant-design/ant-design/issues/18702

**Pull request merged**